cantools/database/utils.py

```diff
@@ -64,12 +63,16 @@ def _encode_fields(fields: Sequence[Union["Signal", "Data"]],
                 if field.is_float:
                     unpacked[field.name] = (value - field.offset) / field.scale
                     continue
-                else:
-                    decimal = (Decimal(value) - Decimal(field.offset)) / Decimal(field.scale)
-                    unpacked[field.name] = int(decimal.to_integral_value())
+
+                if field.offset == 0 and field.scale == 1:
+                    # treat special case to avoid introduction of unnecessary rounding error
+                    unpacked[field.name] = round(value)
```

Comment:
now this is (probably) only correct if `dtype` == `int`? I guess the simplest fix is to introduce 
```python
convert = float if field.is_float else round
```
possibly you can come up with a better name?

Reply:
fixed